packages/apps/plugins/plugin-navtree/src/NavTreePlugin.tsx

```diff
@@ -36,39 +36,48 @@ import {
 import { CommandsTrigger } from './components/CommandsTrigger';
 import meta, { KEY_BINDING, NAVTREE_PLUGIN } from './meta';
 import translations from './translations';
+import { expandOpenGraphNodes, getActions, getChildren, type NavTreeItem, type NavTreeItemGraphNode } from './util';
 
 export type NavTreePluginProvides = SurfaceProvides &
   MetadataRecordsProvides &
   GraphBuilderProvides &
   TranslationsProvides;
 
 export const NavTreePlugin = (): PluginDefinition<NavTreePluginProvides> => {
-  // TODO(wittjosiah): This doesn't support multiple paths for the same node.
-  //   Navtree should base active node on just the id rather than the path.
-  const paths = new Map<string, string[]>();
-  const state = create<{ root?: TreeNode }>({});
+  const state = create<{ root?: NavTreeItemGraphNode; openItemIds?: OpenItemIds }>({});
+
   let graphPlugin: Plugin<GraphProvides> | undefined;
+  let graph: Graph | undefined;
+
+  const handleOpenItemIdsChange = (nextOpenItemIds: OpenItemIds) => {
+    state.openItemIds = nextOpenItemIds;
+    if (graph) {
+      void expandOpenGraphNodes(graph, nextOpenItemIds);
+    }
+  };
 
   return {
     meta,
     ready: async (plugins) => {
       graphPlugin = resolvePlugin(plugins, parseGraphPlugin);
-      const graph = graphPlugin?.provides.graph;
+      graph = graphPlugin?.provides.graph;
       if (!graph) {
         return;
       }
 
-      state.root = treeNodeFromGraphNode(graph, graph.root);
-      state.root.loadChildren();
-      state.root.loadActions();
+      state.root = graph.root as NavTreeItemGraphNode;
+      getChildren(graph, state.root);
+      getActions(graph, state.root);
+
+      // TODO(thure): Do this dynamically.
+      state.openItemIds = { root: true, 'dxos.org/plugin/space-spaces': true, 'dxos.org/plugin/files': true };
```

Comment:
Don't need to block on this but maybe leave a note with an idea of what to try while we have context booted. I'm thinking it could potentially be related to the navtree generator? If it has a callback when it visited each node then you could keep track of the first time you see each node and add nodes with path length of 1 to the open list the first time you see them